pkg/trace/api/profiles.go

```diff
@@ -173,27 +316,30 @@ func (m *multiTransport) RoundTrip(req *http.Request) (rresp *http.Response, rer
 			rresp.StatusCode = http.StatusOK
 		}
 	}()
+
 	if len(m.targets) == 1 {
-		setTarget(req, m.targets[0], m.keys[0])
-		return m.rt.RoundTrip(req)
+		return retryRoundTrip(req, m.targets[0], m.keys[0])
 	}
+
+	// Multiple targets - read body once for all requests
 	slurp, err := io.ReadAll(req.Body)
```

Comment:
It seems like the retry function here is duplicating the reading of the body here?